src/vivarium_ciff_sam/components/lbwsg.py

```diff
@@ -59,23 +65,22 @@ def get_exposure_pipeline(self, builder: Builder) -> Pipeline:
     def get_lbwsg_exposure_pipeline(self, builder: Builder) -> Pipeline:
         return builder.value.get_value(self.lbwsg_exposure_pipeline_name)
 
-    def get_category_endpoints(self, builder: Builder) -> Dict[str, Tuple[float, float]]:
-        category_endpoints = {cat: self.parse_description(description)
-                              for cat, description
-                              in builder.data.load(f'risk_factor.{data_keys.LBWSG.name}.categories').items()}
-        return category_endpoints
+    @classmethod
+    def get_category_intervals(cls, builder: Builder) -> pd.Series:
+        return cls.get_intervals_from_categories(builder.data.load(f'risk_factor.{data_keys.LBWSG.name}.categories'))
 
     ##################################
     # Pipeline sources and modifiers #
     ##################################
 
-    def get_current_exposure(self, index):
+    def get_current_exposure(self, index: pd.Index) -> pd.Series:
         propensities = self.propensity(index)
         lbwsg_categories = self.lbwsg_exposure(index)
 
         def get_exposure_from_category(row: pd.Series) -> float:
-            category_endpoints = self.category_endpoints[row[lbwsg_categories.name]]
-            exposure = row[propensities.name] * (category_endpoints[1] - category_endpoints[0]) + category_endpoints[0]
+            category_interval = self.category_intervals[row[lbwsg_categories.name]]
+            exposure = (row[propensities.name] * (category_interval.right - category_interval.left)
+                        + category_interval.left)
```

Comment:
Is there a reason you chose to have the get_exposure_from_category nested here versus on its own?  Im guessing it is because you always use it for get_current_exposure but what are the benefits of doing it this way?

Reply:
I would assume because this is only needed once, specifically as an argument to df.apply().

Reply:
Cool thanks.

Reply:
@albrja it makes it clear to the reader that this method is intended only for use in this context. it also keeps this method from cluttering up this class's namespace.